modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java

```diff
@@ -39,18 +39,18 @@
  */
 public class DefaultApplicationPolicyInstance implements ApplicationPolicyInstance {
 
-  private ArtifactContext policyContext;
   private final Application application;
   private final PolicyTemplate template;
   private final PolicyParametrization parametrization;
   private final ServiceRepository serviceRepository;
   private final ClassLoaderRepository classLoaderRepository;
   private final List<ArtifactPlugin> artifactPlugins;
   private final ExtensionModelLoaderRepository extensionModelLoaderRepository;
+  private ArtifactContext policyContext;
   private PolicyInstance policyInstance;
 
   /**
-   * Creates a new policy instance
+   * Creates and initialises a new policy instance
```

Comment:
It would be better to remove the initialization logic from the constructor. Add install/init/start/stop methods as in the ones in DeployableArtifact (maybe refactoring and implementing that interface)